beater/api/config/agent/handler.go

```diff
@@ -34,119 +35,109 @@ import (
 
-	msgInvalidQuery               = "invalid query"
-	msgKibanaDisabled             = "disabled Kibana configuration"
-	msgKibanaVersionNotCompatible = "not a compatible Kibana version"
-	msgMethodUnsupported          = "method not supported"
-	msgNoKibanaConnection         = "unable to retrieve connection to Kibana"
-	msgServiceUnavailable         = "service unavailable"
+	msgInvalidQuery       = "invalid query"
+	msgMethodUnsupported  = "method not supported"
+	msgNoKibanaConnection = "unable to retrieve connection to Kibana"
+	msgServiceUnavailable = "service unavailable"
 )
 
 var (
 	// MonitoringMap holds a mapping for request.IDs to monitoring counters
 	MonitoringMap = request.DefaultMonitoringMapForRegistry(registry)
 	registry      = monitoring.Default.NewRegistry("apm-server.acm")
 
-	errMsgKibanaDisabled     = errors.New(msgKibanaDisabled)
-	errMsgNoKibanaConnection = errors.New(msgNoKibanaConnection)
-	errCacheControl          = fmt.Sprintf("max-age=%v, must-revalidate", errMaxAgeDuration.Seconds())
+	errCacheControl = fmt.Sprintf("max-age=%v, must-revalidate", errMaxAgeDuration.Seconds())
 
 	// rumAgents keywords (new and old)
 	rumAgents = []string{"rum-js", "js-base"}
 )
 
-// Handler returns a request.Handler for managing agent central configuration requests.
-func Handler(client kibana.Client, config *config.AgentConfig, defaultServiceEnvironment string) request.Handler {
-	cacheControl := fmt.Sprintf("max-age=%v, must-revalidate", config.Cache.Expiration.Seconds())
-	fetcher := agentcfg.NewFetcher(client, config.Cache.Expiration)
-
-	return func(c *request.Context) {
-		// error handling
-		c.Header().Set(headers.CacheControl, errCacheControl)
+type handler struct {
+	f fetcher
 
-		ok := c.RateLimiter == nil || c.RateLimiter.Allow()
-		if !ok {
-			c.Result.SetDefault(request.IDResponseErrorsRateLimit)
-			c.Write()
-			return
-		}
+	cacheControl, defaultServiceEnvironment string
+}
 
-		if valid := validateClient(c, client, c.AuthResult.Authorized); !valid {
-			c.Write()
-			return
-		}
+func NewHandler(f fetcher, config *config.KibanaAgentConfig, defaultServiceEnvironment string) request.Handler {
+	if f == nil {
+		panic("fetcher must not be nil")
+	}
+	cacheControl := fmt.Sprintf("max-age=%v, must-revalidate", config.Cache.Expiration.Seconds())
+	h := &handler{
+		f:                         f,
+		cacheControl:              cacheControl,
+		defaultServiceEnvironment: defaultServiceEnvironment,
+	}
 
-		query, queryErr := buildQuery(c)
-		if queryErr != nil {
-			extractQueryError(c, queryErr, c.AuthResult.Authorized)
-			c.Write()
-			return
-		}
-		if query.Service.Environment == "" {
-			query.Service.Environment = defaultServiceEnvironment
-		}
+	return h.Handle
+}
 
-		result, err := fetcher.Fetch(c.Request.Context(), query)
-		if err != nil {
-			apm.CaptureError(c.Request.Context(), err).Send()
-			extractInternalError(c, err, c.AuthResult.Authorized)
-			c.Write()
-			return
-		}
+type fetcher interface {
```

Comment:
Seeing as it crosses package boundaries, can we unexport this please? Maybe define it in `agentcfg`?

Reply:
I'll move it over there, but then wouldn't it have to be exported in order to be listed as an argument?

Reply:
I guess you figured it out already, but I mistyped - I meant "export" rather than "unexport". Looks good now, thanks